### ISSUE TITLE
flash kasli-soc fails to find ARM

### ISSUE THREAD

**Issue by jbqubit** [m-labs/artiq#2579]
Looks like JTAG boundary scan fails to find the ARM core on Kasli SoC. 

* Kasli-SOC v1.1.1
* Ubuntu 2022.04 LTS
* Ubuntu AppArmor disabled
* [Incremental build](https://m-labs.hk/artiq/manual/building_developing.html#zynq-incremental-build) of Kasli-SoC using release-8 
* The USB cable is fine -- it worked for this board 1/2024. Other USB cables give the same result. 

```
$ ./local_run.sh   ** with added debug info **
+ '[' -z /nix/store/0wgl1kcha6gh1z9brvg8vnrp4p7dwp88-source/openocd ']'
+ '[' -z /nix/store/rzpgcdhmclwlj275rydxxfqs55izay9z-szl ']'
+ impure=0
+ load_bitstream=1
+ board_type=kasli_soc
+ fw_type=runtime
+ getopts ilb:t:f: opt
+ '[' -z '' ']'
+ case $board_type in
+ board_host=192.168.1.44
+ load_bitstream_cmd=
++ pwd
+ build_dir=/home/britton/experiment/artiq-zynq/build
++ pwd
+ result_dir=/home/britton/experiment/artiq-zynq/result
+ cd /nix/store/0wgl1kcha6gh1z9brvg8vnrp4p7dwp88-source/openocd
+ openocd -f kasli_soc.cfg -c 'load_image /nix/store/rzpgcdhmclwlj275rydxxfqs55izay9z-szl/szl-kasli_soc.elf; resume 0; exit'
Open On-Chip Debugger 0.12.0
Licensed under GNU GPL v2
For bug reports, read
	http://openocd.org/doc/doxygen/bugs.html
DEPRECATED! use 'ftdi device_desc' not 'ftdi_device_desc'
DEPRECATED! use 'ftdi vid_pid' not 'ftdi_vid_pid'
DEPRECATED! use 'ftdi channel' not 'ftdi_channel'
DEPRECATED! use 'ftdi layout_init' not 'ftdi_layout_init'
Zynq CPU1.
DEPRECATED! use 'ftdi layout_signal' not 'ftdi_layout_signal'
Info : clock speed 1000 kHz
Info : JTAG tap: zynq.tap tap/device found: 0x1372c093 (mfg: 0x049 (Xilinx), part: 0x372c, ver: 0x1)
Info : JTAG tap: zynq.dap tap/device found: 0xffffffff (mfg: 0x7ff (<invalid>), part: 0xffff, ver: 0xf)
Warn : JTAG tap: zynq.dap   	UNEXPECTED: 0xffffffff (mfg: 0x7ff (<invalid>), part: 0xffff, ver: 0xf)
Error: JTAG tap: zynq.dap  expected 1 of 1: 0x4ba00477 (mfg: 0x23b (ARM Ltd), part: 0xba00, ver: 0x4)
Error: Trying to use configured scan chain anyway...
Error: zynq.dap: IR capture error; saw 0x0f not 0x01
Warn : Bypassing JTAG setup events due to errors
Error: Invalid ACK (4) in DAP response



$ python3 -m serial.tools.list_ports -v
/dev/ttyUSB0   	 
	desc: Valon USB UART - Valon USB UART
	hwid: USB VID:PID=0403:6001 SER=12202221 LOCATION=1-7.2
/dev/ttyUSB1   	 
	desc: Quad RS232-HS - Quad RS232-HS
	hwid: USB VID:PID=0403:6011 SER=umd-ksoc-001 LOCATION=1-10.4.3:1.0
/dev/ttyUSB2   	 
	desc: Quad RS232-HS - Quad RS232-HS
	hwid: USB VID:PID=0403:6011 SER=umd-ksoc-001 LOCATION=1-10.4.3:1.1
/dev/ttyUSB3   	 
	desc: Quad RS232-HS - Quad RS232-HS
	hwid: USB VID:PID=0403:6011 SER=umd-ksoc-001 LOCATION=1-10.4.3:1.2
/dev/ttyUSB4   	 
	desc: Quad RS232-HS - Quad RS232-HS
	hwid: USB VID:PID=0403:6011 SER=umd-ksoc-001 LOCATION=1-10.4.3:1.3
```

**Comment by jbqubit** [comment]
Related to https://github.com/m-labs/artiq/issues/2250

**Comment by jbqubit** [comment]
I confirmed that the jumpers are correctly positioned and the kasli_soc_por.py was run correctly. The kasli-soc in question still failed as above and the ``FPGA done`` LED does not illuminate. The device moreover does not successfuly boot from a [boot.bin](https://m-labs.hk/artiq/manual/building_developing.html#zynq-incremental-build) loaded onto the flash card. 

Using the same PC with the same nix environment I was able to successfully boot a different kasli-soc using JTAG over USB. 

I conclude that this is a hardware problem with first kasli-soc.
